django/db/backends/sqlite3/schema.py

```diff
@@ -99,6 +99,14 @@ def alter_db_table(self, model, old_db_table, new_db_table, disable_constraints=
             super().alter_db_table(model, old_db_table, new_db_table)
 
     def alter_field(self, model, old_field, new_field, strict=False):
+        # Don't alter if only field name has changed.
+        if (
+            old_field.name != new_field.name and
+            old_field.column == new_field.column and
+            old_field.deconstruct()[1:] == new_field.deconstruct()[1:]
+        ):
```

Comment:
We could reuse `_field_should_be_altered()`:
```suggestion
        if not self._field_should_be_altered(old_field, new_field):
```